tests/unit/python/execution_tree/set_operation.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2018 Steven R. Brandt
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+from phylanx.ast import Phylanx
+import numpy as np
+
+
+# This function is never called, it just makes Flake 8 quiet
+def shape(a, b):
+    pass
+
+
+@Phylanx()
+def kernel(a, b):
+    for i in range(1, shape(a, 0) - 1):
```

Comment:
I think you should be able to use `np.shape(a)[0]` here. This might be closer to what we want to achieve.